rfcs/text/0428-extending-types.md

```diff
@@ -0,0 +1,925 @@
+- Feature Name: extending-and-duplicating-types
```

Comment:
We've discussed calling switching to calling this Inherts/Forking on the top level. Commenting for visibility and later todo.

Reply:
1848a05f915bf5d40c838f9799848798ed144882 and 78a8f596527b051331c81fe72e69e8fdfcdf26f1